master/buildbot/steps/transfer.py

```diff
@@ -231,7 +231,7 @@ def runTransferCommand(self, cmd, writer=None):
 
         @d.addCallback
         def checkResult(_):
-            if cmd.didFail():
+            if writer and cmd.didFail():
```

Comment:
I was mimicking the line below ("if reader and ... ") so just kept the same pattern .. but yes, I think it can only be None for the "not X" case.